modules/apps/forms-and-workflow/calendar/calendar-service/src/main/java/com/liferay/calendar/internal/upgrade/v1_0_6/UpgradeResourcePermissions.java

```diff
@@ -0,0 +1,194 @@
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.StringBundler;
+import com.liferay.portal.security.permission.ResourceActionsImpl;
+import com.liferay.portal.util.PropsValues;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+import java.util.List;
+
+/**
+ * @author José María Muñoz
+ * @author Alberto Chaparro
+ */
+public class UpgradeResourcePermissions extends UpgradeProcess {
+
+	public UpgradeResourcePermissions(
+		ResourceActionLocalService resourceActionLocalService,
+		ResourceBlockLocalService resourceBlockLocalService,
+		RoleLocalService roleLocalService) {
+
+		_resourceActionLocalService = resourceActionLocalService;
+		_resourceBlockLocalService = resourceBlockLocalService;
+		_roleLocalService = roleLocalService;
+	}
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		upgradeGuestResourcePermissions();
+	}
+
+	protected long getCalendarResourceUnsupportedActionsBitwiseValue()
+		throws PortalException {
+
+		int unsupportedBitwiseValue = 0;
+
+		List<String> guestUnsupportedActions =
+			getModelResourceGuestUnsupportedActions();
+
+		for (String resourceActionId : _NEW_UNSUPPORTED_ACTION_IDS) {
+			if (guestUnsupportedActions.contains(resourceActionId)) {
+				ResourceAction resourceAction =
+					_resourceActionLocalService.getResourceAction(
+						_CALENDAR_RESOURCE_NAME, resourceActionId);
+
+				unsupportedBitwiseValue |= resourceAction.getBitwiseValue();
+			}
+		}
+
+		return unsupportedBitwiseValue;
+	}
+
+	protected List<String> getModelResourceGuestUnsupportedActions()
+		throws UpgradeException {
+
+		try {
+			ResourceActionsImpl resourceActionsImpl = new ResourceActionsImpl();
+
+			Class<?> clazz = getClass();
+
+			ClassLoader classLoader = clazz.getClassLoader();
+
+			for (String config : PropsValues.RESOURCE_ACTIONS_CONFIGS) {
+				resourceActionsImpl.read(null, classLoader, config);
+			}
+
+			return resourceActionsImpl.getModelResourceGuestUnsupportedActions(
+				_CALENDAR_RESOURCE_NAME);
+		}
+		catch (Exception e) {
+			throw new UpgradeException(e);
+		}
+	}
+
+	protected void upgradeGuestResourcePermissions() throws Exception {
+		long unsupportedBitwiseValue =
+			getCalendarResourceUnsupportedActionsBitwiseValue();
+
+		if (unsupportedBitwiseValue == 0) {
+			return;
+		}
+
+		StringBundler sb = new StringBundler(2);
+
+		sb.append("select companyId, groupId, calendarResourceId, ");
+		sb.append("resourceBlockId from CalendarResource");
+
+		try (PreparedStatement ps =
+				connection.prepareStatement(sb.toString())) {
+
+			try (ResultSet rs = ps.executeQuery()) {
+				while (rs.next()) {
+					long companyId = rs.getLong(1);
+					long groupId = rs.getLong(2);
+					final long primaryKey = rs.getLong(3);
+					final long resourceBlockId = rs.getLong(4);
+
+					Role guestRole = _roleLocalService.getRole(
+						companyId, RoleConstants.GUEST);
+
+					PermissionedModel permissionedModel =
```

Comment:
Hello @brandizzi!,

Thanks for reviewing! We are not using a SQL statement because those tables (Block and BlockPermission) are from 'core', so, it's suppose that we don't know about how is the implementation to use this tables (because we are implementing an upgrade in CalendarResource Service module and we only should use SQL statement for CalendarResource tables).

Since we can not use SQL statement, we need to instantiate `PermissionedModel` in order to call updateIndividualScopePermissions method form _resourceBlockLocalService.
This solution was comment with @achaparro and @csierra.

Thanks again!